File: rust/src/models/product_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{pagination_response::PaginationResponse, product::Product};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ProductListResponse {
    pub data: Vec<Product>,

    pub pagination_meta: PaginationResponse,
}

impl ProductListResponse {
    pub fn new(data: Vec<Product>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/productListResponse.ts
// this file is @generated
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";
import { type Product, ProductSerializer } from "./product";

export interface ProductListResponse {
  data: Product[];

  paginationMeta: PaginationResponse;
}

export const ProductListResponseSerializer = {
  _fromJsonObject(object: any): ProductListResponse {
    return {
      data: object["data"].map((item: any) => ProductSerializer._fromJsonObject(item)),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: ProductListResponse): any {
    return {
      data: self.data.map((item: any) => ProductSerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
